python/lsst/daf/butler/core/storageClass.py

```diff
@@ -28,11 +28,51 @@
 
 class StorageClassMeta(type):
 
+    # Caches of imported code objects.
+    _pytype = None
+    _assembler = None
+    _disassembler = None
+
     def __init__(self, name, bases, dct):
         super().__init__(name, bases, dct)
         if hasattr(self, "name"):
             StorageClass.subclasses[self.name] = self
 
+    @property
+    def pytype(cls):  # noqa N805
+        if cls._pytype is not None:
+            return cls._pytype
+        # Handle case where we did get a python type not string
+        if not isinstance(cls._pytypeName, str):
+            pytype = cls._pytypeName
+            cls._pytypeName = cls._pytypeName.__name__
+        elif cls._pytypeName == "dict":
+            pytype = dict
+        elif cls._pytypeName == "list":
+            pytype = list
```

Comment:
We need a special case because the `else` clause tries to import them and you can't `import dict`. I hadn't thought about using `dir(builtins)` and seeing if the item exists there and checking it's a type and not a builtin_function_or_method.